contributing/topics/guide-state-migrations.md

```diff
@@ -0,0 +1,170 @@
+# Guide: State Migrations
+
+State migrations come into play if a resource's implementation needs to change, this can happen for a number a reasons, such as the implementation being incorrect or the API that the resource interacts with changes.
+
+Common scenarios where a state migration would be required in Azure are:
+* To correct the format of a Resource ID, the most common example is updating the casing of a segment e.g. `/subscriptions/12345678-1234-9876-4563-123456789012/resourcegroups/resGroup1` -> `/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1`
+* Updating the default value of a property in the schema
+* Recasting property values in the schema
```

Comment:
perhaps worth a disclaimer this should only be done in a major version of the provider/to consider the implications to users configurations too?